From ef94fbc9bb69d73e7ae17f98a172a84759870bdf Mon Sep 17 00:00:00 2001
From: Nicolas Boichat <drinkcat@google.com>
Date: Wed, 8 Jun 2016 23:59:46 +0800
Subject: [PATCH 604/607] platform_android: Add support for DRI Query Buffer
 extension

CQ-DEPEND=CL:*262435
BUG=chrome-os-partner:54012
TEST=Boot to UI.

Change-Id: I0f6d4be3537795c0cae3bf13d6d00b2d40d57e7c
---
 src/egl/drivers/dri2/platform_android.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/egl/drivers/dri2/platform_android.c b/src/egl/drivers/dri2/platform_android.c
index 54c01a4..c20d1b9 100644
--- a/src/egl/drivers/dri2/platform_android.c
+++ b/src/egl/drivers/dri2/platform_android.c
@@ -522,6 +522,9 @@ droid_swap_buffers(_EGLDriver *drv, _EGLDisplay *disp, _EGLSurface *draw)
 
    (*dri2_dpy->flush->invalidate)(dri2_surf->dri_drawable);
 
+   if (dri2_dpy->query_buffers)
+      (*dri2_dpy->query_buffers->query_buffers)(dri2_surf->dri_drawable);
+
    return EGL_TRUE;
 }
 
@@ -1049,7 +1052,8 @@ dri2_initialize_android(_EGLDriver *drv, _EGLDisplay *dpy)
    }
    dri2_dpy->extensions[1] = &use_invalidate.base;
    dri2_dpy->extensions[2] = &image_lookup_extension.base;
-   dri2_dpy->extensions[3] = NULL;
+   dri2_dpy->extensions[3] = &use_query_buffers.base;
+   dri2_dpy->extensions[4] = NULL;
 
 
    if (!dri2_create_screen(dpy)) {
-- 
2.8.0.rc3.226.g39d4020

